lib/site.plugin.js

```diff
@@ -1,38 +1,138 @@
 /**
- * Module variables
+ * Module vars
  */
 const root = '/sites';
 
 class SitePlugin {
+
 	/**
 	 * `SitePlugin` constructor.
 	 *
-	 * @param {String} [id] - the plugin ID
+	 * @param {String} [slug] - the plugin slug
 	 * @param {Number|String} sid - site identifier
 	 * @param {WPCOM} wpcom - wpcom instance
 	 * @return {Undefined} undefined
 	 */
-	constructor( id, sid, wpcom ) {
+	constructor( slug, sid, wpcom ) {
 		if ( ! ( this instanceof SitePlugin ) ) {
-			return new SitePlugin( id, sid, wpcom );
+			return new SitePlugin( slug, sid, wpcom );
+		}
+
+		if ( ! slug ) {
+			throw new Error( '`slug` is not correctly defined' );
 		}
 
-		this._id = id;
+		this._slug = encodeURIComponent( slug );
 		this._sid = sid;
-		this.path = `${root}/${this._sid}/plugins`;
 		this.wpcom = wpcom;
+
+		const path = `${root}/${ this._sid }/plugins`;
```

Comment:
I did it in my way :-o
I replaced `this.path` by `const path`. Let me remove it completely.